tests/test_flow.py

```diff
@@ -3545,6 +3632,39 @@ def test_none_key_date_is_not_written():
     assert result == expected
 
 
+def test_str_id_with_no_header():
+    with pytest.raises(ValueError) as value_error:
+        build_flow(
+            [
+                CSVSource("tests/test.csv", header=False, id_field="n1"),
+            ]
+        ).run()
+
+    assert str(value_error.value) == "id_field can only be set to an integer when with_header is false"
+
+
+def test_str_key_field_with_no_header():
+    with pytest.raises(ValueError) as value_error:
+        build_flow(
+            [
+                CSVSource("tests/test.csv", header=False, key_field="n1"),
+            ]
+        ).run()
+
+    assert str(value_error.value) == "key_field can only be set to an integer when with_header is false"
+
+
+def test_mixed_key_types():
+    with pytest.raises(ValueError) as value_error:
+        build_flow(
+            [
+                CSVSource("tests/test.csv", header=True, key_field=["n1", 2]),
+            ]
+        ).run()
+
+    assert str(value_error.value) == "Keys should not contain both integer and string values"
```

Comment:
To be consistent with other error messages:
```suggestion
    assert str(value_error.value) == "key_field should not contain both integer and string values"
```